src/brsTypes/index.ts

```diff
@@ -64,24 +66,39 @@ export function isBrsCallable(value: BrsType): value is Callable {
  * @returns `true` if `value` can be iterated across, otherwise `false`.
  */
 export function isIterable(value: BrsType): value is Iterable {
-    return value.kind === ValueKind.Array;
+    switch (value.kind) {
+        case ValueKind.Array:
+        case ValueKind.AssociativeArray:
+            return true;
+        default:
+            return false;
+    }
+}
+
+/**
+ * Determines whether ot not the provided value is a primitive BrightScript
+ * value, i.e. one of:
+ *   * `invalid`
+ *   * `true` or `false`
+ *   * any numeric value
+ *   * any string
+ */
+export function isBrsPrimitive(value: BrsType): value is BrsPrimitive {
+    return value.kind < ValueKind.Array;
 }
 
 /** The set of BrightScript numeric types. */
 export type BrsNumber = Int32 | Int64 | Float | Double;
 
+/** The set of all primitive (i.e. intrinsic and unboxed) BrightScript types. */
+export type BrsPrimitive = BrsInvalid | BrsBoolean | BrsString | BrsNumber;
+
 /** The set of BrightScript iterable types. */
-export type Iterable = BrsArray;
+export type Iterable = BrsArray | AssociativeArray;
```

Comment:
Within the `brsTypes` tree, the `Brs` prefix is there to avoid name collisions with JavaScript's `Array`, `Boolean`, and `String` types.  BrightScript's `invalid` got the `Brs` prefix to make it more clear that `Invalid` is a BrightScript type, since it's otherwise not super obvious.  `AssociativeArray`, `Int32`, `Float`, `Double`, and `Iterable` got a pass because there's very little overlap between those and JavaScript.

Outside of that tree, I'd like to eventually modify how things are exported to prefer using these types as `Brs.Array`, `Brs.String`, `Brs.Int32` etc.  Or something like that.  But that's a long way off.